### DISCUSSION TITLE
Is it normal for Homebrew to install under /usr/local on M1 Mac?

### DISCUSSION THREAD

**Discussion by 0x5am5** [Homebrew/discussions#3055] [Getting started] [answered]
This is all rooted in the fact I cannot install Ruby with asdf. I'm trying to find the source of this issue.

Discussions like [this](https://github.com/rbenv/ruby-build/issues/1691#issuecomment-983122764) suggest reinstalling Homebrew.

They all seem to talk about M1 installing under `opt/homebrew` but when I install Homebrew it gives me the following message which doesn't match what my expectation is (to install under opt/homebrew. 

```
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> The following new directories will be created:
/usr/local/Cellar
/usr/local/Caskroom
```

When I moved over to M1 I did a transfer from Intel (and not from a fresh install) I wonder if all these issues are because of this.

**Comment by SMillerDev** [comment]
If you install with rosetta brew will install in the default intel prefix (`/usr/local`).

**Reply by 0x5am5** [reply]
Thank you for taking the time to answer! I really appreciate it. I will say however, this was a pretty unhelpful response as it didn't really answer my question.

**Comment by gromgit** [comment, marked as answer]
That means you're running the Homebrew installer in a _Rosetta-enabled Terminal_, so the installer switches to "Intel mode", installing Intel binaries under `/usr/local`.

You should launch a fresh Terminal session, then run the installer from there, to get "native mode" which installs AS binaries under `/opt/homebrew`. Alternatively, you should be able to "invoke" native mode in a Rosetta Terminal with:
```
arch -arm64e /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Reply by 0x5am5** [reply]
This was perfect. I think VS Code runs its terminal under rosetta which is why this was happening.

Your command works but alternatively (its hard to say this for certain) I think I should have tried installing in Terminal and maybe the arm64 prefix would not be needed.

**Reply by 0x5am5** [reply]
Continuing my investigation, I believe this is exactly what the issue was. My solution was to download the Apple Silicon Version of VSCode.

https://code.visualstudio.com/Download#
